### PR TITLE
Added UserDTO

### PR DESCRIPTION
**Pull request by Elm0xz**
Not much to check, added DTO for User to decouple domain objects from json operations.

https://trello.com/c/wOkvZoUw/20-introduce-dtos-to-server-logic

### CODE REVIEW

File: src/main/java/com/pretz/dyntreeserver/domain/User.java [hunk 1]
```diff
@@ -26,6 +24,11 @@
     public User() {
     }
 
+    public User(UserDTO userDTO) {
```
Comment (lgodziejewski) [inline src/main/java/com/pretz/dyntreeserver/domain/User.java]: hmm, I'm not sure `User` should depend on `UserDTO`.

Maybe add a method to `UserDTO` called `toModel()`, containing sth like:

```
public User toModel() {
  return new User(this.fieldA, this.fieldB);
}
```
?

just a suggestion.

Or even better - don't create dependencies in any of these files and create an external "translator" utility which does DTO <-> Model mapping?

File: src/main/java/com/pretz/dyntreeserver/service/dto/UserDTO.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package com.pretz.dyntreeserver.service.dto;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class UserDTO {
+
+    private String name;
+
+    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
```
Comment (lgodziejewski) [inline src/main/java/com/pretz/dyntreeserver/service/dto/UserDTO.java:10]: what's this for?

File: src/main/java/com/pretz/dyntreeserver/service/dto/UserDTO.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    @Override
+    public String toString() {
```
Comment (lgodziejewski) [inline src/main/java/com/pretz/dyntreeserver/service/dto/UserDTO.java:50]: is it necessary? looks like default implementation (except for the lack of `password` field)

Comment (Elm0xz) [reply src/main/java/com/pretz/dyntreeserver/service/dto/UserDTO.java:50]: I'm planning to use it in unit tests

File: src/main/java/com/pretz/dyntreeserver/service/dto/UserDTO.java (hunk 2 above)
Comment (Elm0xz) [reply src/main/java/com/pretz/dyntreeserver/service/dto/UserDTO.java:10]: Without it, JSON parser ignored password field on both serialization and deserialization. In effect, it didn't read password parameter from JSON and what I got were domain objects with null password. Not pretty

File: src/main/java/com/pretz/dyntreeserver/domain/User.java (hunk 1 above)
Comment (Elm0xz) [reply src/main/java/com/pretz/dyntreeserver/domain/User.java]: I can use normal constructor and pass name and email from DTO as parameters.

**Review by lgodziejewski — APPROVED** [review]
Looks good as for now, but eventually I'd introduce a Model <-> Dto mapping mechanism

File: src/main/java/com/pretz/dyntreeserver/service/dto/UserDTO.java (hunk 3 above)
Comment (lgodziejewski) [reply src/main/java/com/pretz/dyntreeserver/service/dto/UserDTO.java:50]: Code should not contain logic prepared specifically for unit tests :-)

Comment (Elm0xz) [reply src/main/java/com/pretz/dyntreeserver/service/dto/UserDTO.java:50]: It's only toString, please don't be so papal

**Pull request merged**